ui/glide/src/main/java/de/danoeh/antennapod/ui/glide/ImageModelLoader.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    @Override
+    public boolean handles(@NonNull ImageModel model) {
+        return true;
+    }
+
+    static class ImageModelFetcher implements DataFetcher<InputStream> {
+        private static final int[] PALETTES = {0xff78909c, 0xffff6f00, 0xff388e3c,
+                0xff00838f, 0xff7b1fa2, 0xffb71c1c, 0xff2196f3};
+        
+        private final ImageModel model;
+        private final int width;
+        private final int height;
+        private final Context context;
+
+        public ImageModelFetcher(ImageModel model, int width, int height, Context context) {
+            this.model = model;
+            this.width = width;
+            this.height = height;
+            this.context = context;
+        }
+
+        @Override
+        public void loadData(@NonNull Priority priority, @NonNull DataCallback<? super InputStream> callback) {
+            try {
+                boolean triedNetworkUrl = false;
+                
+                // Try primary URL first
+                if (model.hasPrimaryUrl()) {
+                    triedNetworkUrl = true;
+                    InputStream stream = loadFromUrl(model.getPrimaryUrl());
+                    if (stream != null) {
+                        callback.onDataReady(stream);
+                        return;
+                    }
+                }
+                
+                // Try fallback URL
+                if (model.hasFallbackUrl()) {
+                    triedNetworkUrl = true;
+                    InputStream stream = loadFromUrl(model.getFallbackUrl());
+                    if (stream != null) {
+                        callback.onDataReady(stream);
+                        return;
+                    }
+                }
+                
+                if (triedNetworkUrl) {
+                    // If we tried network URLs but they failed, report as network error
+                    callback.onLoadFailed(new Exception("Network URLs failed"));
+                } else {
+                    // icon is not defined, generate a placeholder with fallback text
+                    callback.onDataReady(generatePlaceholder());
+                }
+            } catch (RuntimeException e) {
+                callback.onLoadFailed(e);
+            }
+        }
+
+        private InputStream loadFromUrl(String url) {
+            if (url == null) {
+                return null;
+            }
+            
+            try {
+                // Handle content: URLs
+                if (url.startsWith(ContentResolver.SCHEME_CONTENT)) {
+                    return context.getContentResolver().openInputStream(Uri.parse(url));
+                }
+
+                // Handle HTTP/HTTPS URLs
+                HttpURLConnection connection = (HttpURLConnection) new URL(url).openConnection();
+                connection.setConnectTimeout(5000);
+                connection.setReadTimeout(10000);
+                connection.setInstanceFollowRedirects(true);
```

Comment:
This looks more complex than I expected. Can't we somehow call the existing Glide functionality? We even have some code already that downsamples huge images, and doing the I/O manually here would break that.

Reply:
This will probably still break stuff (interceptors, resizing, content uris). Please try to use ApOkHttpUrlLoader instead somehow.